d2ir/d2ir.go

```diff
@@ -318,11 +318,19 @@ func (f *Field) Copy(newParent Node) Node {
 }
 
 func (f *Field) LastPrimaryRef() Reference {
+	var lastGlobSet Reference
```

Comment:
I think `(e *Edge) LastPrimaryRef()` might also need this if there are multiple globs affecting an edge?

Reply:
hmm adding 

```d2
(** -> **)[*]: {
	label.near: center
}
```

to the test correctly shows the new error, maybe this isn't needed?

Reply:
oh wait

```d2
**: {
  label: hi
  label.near: center
}
(** -> **)[*]: {
	label.near: top-center
}
(** -> **)[*]: {
	label.near: invalid
}

x: {
  a -> b
}
```

has the line number of the first valid near instead of the 2nd
```
        d2/testdata/d2compiler/TestCompile2/globs/double-glob-err-val.d2:4:3: invalid "near" field
        d2/testdata/d2compiler/TestCompile2/globs/double-glob-err-val.d2:7:2: invalid "near" field
```

line 7 should be 10 right?

Reply:
gooood catch